frontend/src/pages/MilestonesEditor.tsx

```diff
@@ -167,52 +160,39 @@ export const MilestonesEditor = () => {
       .to(copyLists[destination.droppableId], destination.index);
     setVersionLists(copyLists);
 
-    if (destination.droppableId !== ROADMAP_LIST_ID) {
+    if (destination.droppableId === ROADMAP_LIST_ID) {
+      // If moving to unversioned-list, just remove it from source version
+      const payload = versionPayload(roadmapsVersions, +source.droppableId);
+      payload.tasks = payload.tasks.filter(
+        (taskId) => taskId !== +result.draggableId,
+      );
+      await patchVersion(payload).unwrap();
+    } else {
       // If moving to another version -> add to new version
+      // Backend deletes tasks from previous version automatically
       const payload = versionPayload(
         roadmapsVersions,
         +destination.droppableId,
       );
       payload.tasks.splice(destination.index, 0, +result.draggableId);
       await patchVersion(payload).unwrap();
     }
-
-    if (source.droppableId !== ROADMAP_LIST_ID) {
-      // If moving from another version -> remove from previous version
-      const payload = versionPayload(roadmapsVersions, +source.droppableId);
-      payload.tasks = payload.tasks.filter(
-        (taskId) => taskId !== +result.draggableId,
-      );
-      await patchVersion(payload).unwrap();
-    }
   };
 
   const onTaskDragEnd = async (result: DropWithDestination) => {
     const { source, destination } = result;
-
-    // Backup list that is not mutated, used for reverting action if api request fails
-    const backupLists = copyVersionLists(versionLists);
```

Comment:
One case where the view can get out of sync is when the browser is offline.
The tasks can be moved and the edits stay locally, but obviously don't go through.

Either way the page needs to be refreshed to get up to date data after getting back to online, so I don't think the backup list is strictly necessary.

But in that case we should at least show errors. And in case of error, the `versionLists` could be reset from the last fetched data (e.i. we could run the `useEffect`).